# Wheeler Wilkinson Project

## Import Libraries
Any libraries we import for data handling will go here.

In [24]:
import pandas as pd

## Load the Dataset
We used pandas to read the CSV file into a DataFrame.

In [26]:
chicago = pd.read_csv('Chicago_Crime.csv')
LA = pd.read_csv('LA_Crime.csv')
print("Dataset loaded successfully.")

Dataset loaded successfully.


### Chicago Information
We printed basic information about the DataFrame, such as its shape, columns, and a preview of the first few rows.

In [27]:
print(f"Dataset shape: {chicago.shape}")
print(f"Columns: {list(chicago.columns)}")

Dataset shape: (212702, 22)
Columns: ['ID', 'Case Number', 'Date', 'Block', 'IUCR', 'Primary Type', 'Description', 'Location Description', 'Arrest', 'Domestic', 'Beat', 'District', 'Ward', 'Community Area', 'FBI Code', 'X Coordinate', 'Y Coordinate', 'Year', 'Updated On', 'Latitude', 'Longitude', 'Location']


In [28]:
chicago.head(5)

,ID,Case Number,Date,Block,IUCR,Primary Type,Description,Location Description,Arrest,Domestic,...,Ward,Community Area,FBI Code,X Coordinate,Y Coordinate,Year,Updated On,Latitude,Longitude,Location
0,12262588,JE104819,12/31/2020 11:59:00 PM,017XX N CICERO AVE,0810,THEFT,OVER $500,PARKING LOT / GARAGE (NON RESIDENTIAL),False,False,...,37.0,25.0,06,1144138.0,1910955.0,2020,2021 Jan 08 03:39:34 PM,41.911684,-87.745918,POINT (-87.745917518 41.9116839)
1,12258658,JE100080,12/31/2020 11:51:00 PM,013XX S HOMAN AVE,0430,BATTERY,AGGRAVATED - OTHER DANGEROUS WEAPON,STREET,False,False,...,24.0,29.0,04B,1153918.0,1893431.0,2020,2021 Jan 07 03:41:47 PM,41.863407,-87.710456,POINT (-87.710455868 41.863406877)
2,12274737,JE118886,12/31/2020 11:45:00 PM,001XX E WACKER DR,0810,THEFT,OVER $500,HOTEL / MOTEL,False,False,...,42.0,32.0,06,1177683.0,1902638.0,2020,2021 Jan 23 03:39:36 PM,41.888165,-87.622937,POINT (-87.622937212 41.888165132)
3,12258623,JD475015,12/31/2020 11:45:00 PM,056XX S WABASH AVE,0486,BATTERY,DOMESTIC BATTERY SIMPLE,APARTMENT,False,True,...,20.0,40.0,08B,1177678.0,1867609.0,2020,2021 Jan 07 03:41:47 PM,41.792043,-87.624018,POINT (-87.624017919 41.792043078)
4,12263896,JE105660,12/31/2020 11:45:00 PM,049XX S FORRESTVILLE AVE,0810,THEFT,OVER $500,RESIDENCE,False,False,...,4.0,38.0,06,1180760.0,1872402.0,2020,2021 Jan 10 03:39:57 PM,41.805125,-87.612569,POINT (-87.612569477 41.805125154)


### LA Information
We printed basic information about the DataFrame, such as its shape, columns, and a preview of the first few rows.

In [29]:
print(f"Dataset shape: {LA.shape}")
print(f"Columns: {list(LA.columns)}")

Dataset shape: (1004894, 28)
Columns: ['DR_NO', 'Date Rptd', 'DATE OCC', 'TIME OCC', 'AREA', 'AREA NAME', 'Rpt Dist No', 'Part 1-2', 'Crm Cd', 'Crm Cd Desc', 'Mocodes', 'Vict Age', 'Vict Sex', 'Vict Descent', 'Premis Cd', 'Premis Desc', 'Weapon Used Cd', 'Weapon Desc', 'Status', 'Status Desc', 'Crm Cd 1', 'Crm Cd 2', 'Crm Cd 3', 'Crm Cd 4', 'LOCATION', 'Cross Street', 'LAT', 'LON']


In [30]:
LA.head(5)

,DR_NO,Date Rptd,DATE OCC,TIME OCC,AREA,AREA NAME,Rpt Dist No,Part 1-2,Crm Cd,Crm Cd Desc,...,Status,Status Desc,Crm Cd 1,Crm Cd 2,Crm Cd 3,Crm Cd 4,LOCATION,Cross Street,LAT,LON
0,211507896,2021 Apr 11 12:00:00 AM,2020 Nov 07 12:00:00 AM,845,15,N Hollywood,1502,2,354,THEFT OF IDENTITY,...,IC,Invest Cont,354.0,NaN,NaN,NaN,7800 BEEMAN AV,NaN,34.2124,-118.4092
1,201516622,2020 Oct 21 12:00:00 AM,2020 Oct 18 12:00:00 AM,1845,15,N Hollywood,1521,1,230,"ASSAULT WITH DEADLY WEAPON, AGGRAVATED ASSAULT",...,IC,Invest Cont,230.0,NaN,NaN,NaN,ATOLL AV,N GAULT,34.1993,-118.4203
2,240913563,2024 Dec 10 12:00:00 AM,2020 Oct 30 12:00:00 AM,1240,9,Van Nuys,933,2,354,THEFT OF IDENTITY,...,IC,Invest Cont,354.0,NaN,NaN,NaN,14600 SYLVAN ST,NaN,34.1847,-118.4509
3,210704711,2020 Dec 24 12:00:00 AM,2020 Dec 24 12:00:00 AM,1310,7,Wilshire,782,1,331,THEFT FROM MOTOR VEHICLE - GRAND ($950.01 AND ...,...,IC,Invest Cont,331.0,NaN,NaN,NaN,6000 COMEY AV,NaN,34.0339,-118.3747
4,201418201,2020 Oct 03 12:00:00 AM,2020 Sep 29 12:00:00 AM,1830,14,Pacific,1454,1,420,THEFT FROM MOTOR VEHICLE - PETTY ($950 & UNDER),...,IC,Invest Cont,420.0,NaN,NaN,NaN,4700 LA VILLA MARINA,NaN,33.9813,-118.4350


# Cleaning

The Chicago dataset "Date" refers to the date and time that the event occured. Therefore we will also use the date and time the event occurd (DATE OCC) from the LA Dataset. Using datetime we formated the DATE OCC to match that of Chicago and droped all row that were not from the year 2020.

In [35]:
LA['DATE OCC'] = pd.to_datetime(LA['DATE OCC'])

LA = LA[LA['DATE OCC'].dt.year == 2020]

# Reformat to match the Chicago format: %m/%d/%Y %I:%M:%S %p
LA['DATE OCC'] = LA['DATE OCC'].dt.strftime('%m/%d/%Y %I:%M:%S %p')

LA.head(5)

/var/folders/3f/pww76vqd6_b67qs48cp32q980000gn/T/ipykernel_43139/2187665707.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  LA['DATE OCC'] = pd.to_datetime(LA['DATE OCC'])


,DR_NO,Date Rptd,DATE OCC,TIME OCC,AREA,AREA NAME,Rpt Dist No,Part 1-2,Crm Cd,Crm Cd Desc,...,Status,Status Desc,Crm Cd 1,Crm Cd 2,Crm Cd 3,Crm Cd 4,LOCATION,Cross Street,LAT,LON
0,211507896,2021 Apr 11 12:00:00 AM,11/07/2020 12:00:00 AM,845,15,N Hollywood,1502,2,354,THEFT OF IDENTITY,...,IC,Invest Cont,354.0,NaN,NaN,NaN,7800 BEEMAN AV,NaN,34.2124,-118.4092
1,201516622,2020 Oct 21 12:00:00 AM,10/18/2020 12:00:00 AM,1845,15,N Hollywood,1521,1,230,"ASSAULT WITH DEADLY WEAPON, AGGRAVATED ASSAULT",...,IC,Invest Cont,230.0,NaN,NaN,NaN,ATOLL AV,N GAULT,34.1993,-118.4203
2,240913563,2024 Dec 10 12:00:00 AM,10/30/2020 12:00:00 AM,1240,9,Van Nuys,933,2,354,THEFT OF IDENTITY,...,IC,Invest Cont,354.0,NaN,NaN,NaN,14600 SYLVAN ST,NaN,34.1847,-118.4509
3,210704711,2020 Dec 24 12:00:00 AM,12/24/2020 12:00:00 AM,1310,7,Wilshire,782,1,331,THEFT FROM MOTOR VEHICLE - GRAND ($950.01 AND ...,...,IC,Invest Cont,331.0,NaN,NaN,NaN,6000 COMEY AV,NaN,34.0339,-118.3747
4,201418201,2020 Oct 03 12:00:00 AM,09/29/2020 12:00:00 AM,1830,14,Pacific,1454,1,420,THEFT FROM MOTOR VEHICLE - PETTY ($950 & UNDER),...,IC,Invest Cont,420.0,NaN,NaN,NaN,4700 LA VILLA MARINA,NaN,33.9813,-118.4350


In [37]:
chicago.shape

(212702, 22)

In [36]:
LA.shape

(199847, 28)

## Looking at Stadardizing the Crime Types for Integration

In [31]:
chicago['Primary Type'].nunique()

33

In [32]:
LA['Crm Cd Desc'].nunique()

140

In [33]:
print(chicago['Primary Type'].unique())

['THEFT' 'BATTERY' 'CRIMINAL DAMAGE' 'WEAPONS VIOLATION'
 'MOTOR VEHICLE THEFT' 'BURGLARY' 'OTHER OFFENSE' 'ROBBERY' 'NARCOTICS'
 'CRIMINAL TRESPASS' 'PUBLIC PEACE VIOLATION' 'ASSAULT'
 'LIQUOR LAW VIOLATION' 'DECEPTIVE PRACTICE' 'STALKING' 'SEX OFFENSE'
 'OFFENSE INVOLVING CHILDREN' 'CRIMINAL SEXUAL ASSAULT' 'ARSON' 'HOMICIDE'
 'INTERFERENCE WITH PUBLIC OFFICER' 'INTIMIDATION'
 'CONCEALED CARRY LICENSE VIOLATION' 'PROSTITUTION' 'KIDNAPPING'
 'OBSCENITY' 'PUBLIC INDECENCY' 'GAMBLING' 'RITUALISM' 'HUMAN TRAFFICKING'
 'OTHER NARCOTIC VIOLATION' 'NON-CRIMINAL' 'CRIM SEXUAL ASSAULT']


In [34]:
print(LA['Crm Cd Desc'].unique())

['THEFT OF IDENTITY' 'ASSAULT WITH DEADLY WEAPON, AGGRAVATED ASSAULT'
 'THEFT FROM MOTOR VEHICLE - GRAND ($950.01 AND OVER)'
 'THEFT FROM MOTOR VEHICLE - PETTY ($950 & UNDER)'
 'CRM AGNST CHLD (13 OR UNDER) (14-15 & SUSP 10 YRS OLDER)'
 'VEHICLE - STOLEN' 'BURGLARY' 'BURGLARY FROM VEHICLE'
 'THEFT PLAIN - PETTY ($950 & UNDER)' 'INTIMATE PARTNER - SIMPLE ASSAULT'
 'BATTERY - SIMPLE ASSAULT' 'VANDALISM - MISDEAMEANOR ($399 OR UNDER)'
 'VEHICLE - ATTEMPT STOLEN'
 'VANDALISM - FELONY ($400 & OVER, ALL CHURCH VANDALISMS)' 'ROBBERY'
 'FIREARMS RESTRAINING ORDER (FIREARMS RO)' 'BIKE - STOLEN'
 'EMBEZZLEMENT, GRAND THEFT ($950.01 & OVER)'
 'CHILD ABUSE (PHYSICAL) - SIMPLE ASSAULT'
 'CRIMINAL THREATS - NO WEAPON DISPLAYED'
 'THEFT-GRAND ($950.01 & OVER)EXCPT,GUNS,FOWL,LIVESTK,PROD'
 'BATTERY WITH SEXUAL CONTACT' 'LETTERS, LEWD  -  TELEPHONE CALLS, LEWD'
 'VIOLATION OF COURT ORDER' 'ARSON' 'VIOLATION OF RESTRAINING ORDER'
 'THEFT, PERSON' 'CONTEMPT OF COURT'
 'INTIMATE PARTNER - AGGRAVATED ASSAU

Chicago and Los Angeles use entirely different local reporting standards. LA breaks things down to a highly specific degree, while Chicago groups much more broadly. To get these two datasets to have a universal crime type system, we used Google Gemini to help us build a mapping dictionary that rolls both of these localized lists up into a unified set of about 15 standard crime categories.

We gave Gemini the full Crime type lists from Chicago (33 different possible values) and LA (140 different possible values). Gemini generated Python code using a dictionary inversion method to recategorize both datasets. We also added a feature to check for a variable that might slip through the cracks of Gemini's dictionary. Any values not explicitly listed will map to "UNMAPPED" so we can check for outliers.


In [42]:
unified_mapping = {
    'HOMICIDE': [
        'HOMICIDE', 'CRIMINAL HOMICIDE', 'MANSLAUGHTER, NEGLIGENT'
    ],
    'ASSAULT & BATTERY': [
        'BATTERY', 'ASSAULT', 'INTIMIDATION', 'STALKING',
        'ASSAULT WITH DEADLY WEAPON, AGGRAVATED ASSAULT', 'INTIMATE PARTNER - SIMPLE ASSAULT', 
        'BATTERY - SIMPLE ASSAULT', 'CHILD ABUSE (PHYSICAL) - SIMPLE ASSAULT', 
        'CRIMINAL THREATS - NO WEAPON DISPLAYED', 'BATTERY WITH SEXUAL CONTACT', 
        'INTIMATE PARTNER - AGGRAVATED ASSAULT', 'BATTERY POLICE (SIMPLE)', 'OTHER ASSAULT', 
        'ASSAULT WITH DEADLY WEAPON ON POLICE OFFICER', 'CHILD ABUSE (PHYSICAL) - AGGRAVATED ASSAULT', 
        'BATTERY ON A FIREFIGHTER', 'THREATENING PHONE CALLS/LETTERS'
    ],
    'SEX OFFENSES': [
        'SEX OFFENSE', 'CRIMINAL SEXUAL ASSAULT', 'CRIM SEXUAL ASSAULT', 'PROSTITUTION', 
        'OBSCENITY', 'PUBLIC INDECENCY',
        'LEWD/LASCIVIOUS ACTS WITH CHILD', 'RAPE, ATTEMPTED', 'SODOMY/SEXUAL CONTACT B/W PENIS OF ONE PERS TO ANUS OTH', 
        'RAPE, FORCIBLE', 'SEX OFFENDER REGISTRANT OUT OF COMPLIANCE', 'SEX,UNLAWFUL(INC MUTUAL CONSENT, PENETRATION W/ FRGN OBJ', 
        'ORAL COPULATION', 'SEXUAL PENETRATION W/FOREIGN OBJECT', 'LEWD CONDUCT', 'INDECENT EXPOSURE', 
        'CHILD PORNOGRAPHY', 'PANDERING', 'PIMPING', 'BEASTIALITY, CRIME AGAINST NATURE SEXUAL ASSLT WITH ANIM', 
        'INCEST (SEXUAL ACTS BETWEEN BLOOD RELATIVES)', 'LETTERS, LEWD  -  TELEPHONE CALLS, LEWD'
    ],
    'CRIMES AGAINST CHILDREN': [
        'OFFENSE INVOLVING CHILDREN',
        'CRM AGNST CHLD (13 OR UNDER) (14-15 & SUSP 10 YRS OLDER)', 'CHILD NEGLECT (SEE 300 W.I.C.)', 
        'CONTRIBUTING', 'CHILD ANNOYING (17YRS & UNDER)', 'CHILD ABANDONMENT'
    ],
    'ROBBERY': [
        'ROBBERY', 'ATTEMPTED ROBBERY', 'PURSE SNATCHING', 'PURSE SNATCHING - ATTEMPT'
    ],
    'BURGLARY': [
        'BURGLARY', 'BURGLARY FROM VEHICLE', 'BURGLARY, ATTEMPTED', 'BURGLARY FROM VEHICLE, ATTEMPTED'
    ],
    'THEFT / LARCENY': [
        'THEFT',
        'THEFT OF IDENTITY', 'THEFT PLAIN - PETTY ($950 & UNDER)', 'THEFT-GRAND ($950.01 & OVER)EXCPT,GUNS,FOWL,LIVESTK,PROD', 
        'THEFT, PERSON', 'SHOPLIFTING - PETTY THEFT ($950 & UNDER)', 'SHOPLIFTING-GRAND THEFT ($950.01 & OVER)', 
        'PICKPOCKET', 'THEFT PLAIN - ATTEMPT', 'SHOPLIFTING - ATTEMPT', 'THEFT, COIN MACHINE - PETTY ($950 & UNDER)', 
        'THEFT FROM PERSON - ATTEMPT', 'THEFT, COIN MACHINE - ATTEMPT', 'PETTY THEFT - AUTO REPAIR', 
        'GRAND THEFT / AUTO REPAIR', 'THEFT, COIN MACHINE - GRAND ($950.01 & OVER)', 'TILL TAP - PETTY ($950 & UNDER)', 
        'PICKPOCKET, ATTEMPT', 'TILL TAP - GRAND THEFT ($950.01 & OVER)', 'DEFRAUDING INNKEEPER/THEFT OF SERVICES, $950 & UNDER', 
        'DEFRAUDING INNKEEPER/THEFT OF SERVICES, OVER $950.01', 'THEFT FROM MOTOR VEHICLE - GRAND ($950.01 AND OVER)', 
        'THEFT FROM MOTOR VEHICLE - PETTY ($950 & UNDER)', 'THEFT FROM MOTOR VEHICLE - ATTEMPT'
    ],
    'MOTOR VEHICLE THEFT': [
        'MOTOR VEHICLE THEFT',
        'VEHICLE - STOLEN', 'VEHICLE - ATTEMPT STOLEN', 'BIKE - STOLEN', 'VEHICLE, STOLEN - OTHER (MOTORIZED SCOOTERS, BIKES, ETC)', 
        'DRIVING WITHOUT OWNER CONSENT (DWOC)', 'BOAT - STOLEN', 'BIKE - ATTEMPTED STOLEN'
    ],
    'DRUGS & NARCOTICS': [
        'NARCOTICS', 'OTHER NARCOTIC VIOLATION',
        'DRUGS, TO A MINOR'
    ],
    'WEAPONS OFFENSES': [
        'WEAPONS VIOLATION', 'CONCEALED CARRY LICENSE VIOLATION',
        'FIREARMS RESTRAINING ORDER (FIREARMS RO)', 'BRANDISH WEAPON', 'DISCHARGE FIREARMS/SHOTS FIRED', 
        'SHOTS FIRED AT MOVING VEHICLE, TRAIN OR AIRCRAFT', 'SHOTS FIRED AT INHABITED DWELLING', 
        'REPLICA FIREARMS(SALE,DISPLAY,MANUFACTURE OR DISTRIBUTE)', 'WEAPONS POSSESSION/BOMBING', 
        'FIREARMS EMERGENCY PROTECTIVE ORDER (FIREARMS EPO)'
    ],
    'FRAUD & WHITE COLLAR': [
        'DECEPTIVE PRACTICE',
        'EMBEZZLEMENT, GRAND THEFT ($950.01 & OVER)', 'BUNCO, ATTEMPT', 'DOCUMENT FORGERY / STOLEN FELONY', 
        'BUNCO, GRAND THEFT', 'BUNCO, PETTY THEFT', 'DISHONEST EMPLOYEE - PETTY THEFT', 'EMBEZZLEMENT, PETTY THEFT ($950 & UNDER)', 
        'CREDIT CARDS, FRAUD USE ($950 & UNDER', 'DOCUMENT WORTHLESS ($200 & UNDER)', 'CREDIT CARDS, FRAUD USE ($950.01 & OVER)', 
        'COUNTERFEIT', 'DISHONEST EMPLOYEE - GRAND THEFT', 'DOCUMENT WORTHLESS ($200.01 & OVER)', 'GRAND THEFT / INSURANCE FRAUD', 
        'DISHONEST EMPLOYEE ATTEMPTED THEFT', 'BRIBERY', 'EXTORTION', 'UNAUTHORIZED COMPUTER ACCESS'
    ],
    'VANDALISM & PROPERTY DAMAGE': [
        'CRIMINAL DAMAGE', 'ARSON',
        'VANDALISM - MISDEAMEANOR ($399 OR UNDER)', 'VANDALISM - FELONY ($400 & OVER, ALL CHURCH VANDALISMS)', 
        'THROWING OBJECT AT MOVING VEHICLE', 'TELEPHONE PROPERTY - DAMAGE', 'ILLEGAL DUMPING'
    ],
    'PUBLIC ORDER & TRESPASS': [
        'CRIMINAL TRESPASS', 'PUBLIC PEACE VIOLATION', 'LIQUOR LAW VIOLATION', 'INTERFERENCE WITH PUBLIC OFFICER',
        'VIOLATION OF COURT ORDER', 'VIOLATION OF RESTRAINING ORDER', 'CONTEMPT OF COURT', 'TRESPASSING', 'FAILURE TO YIELD', 
        'RESISTING ARREST', 'VIOLATION OF TEMPORARY RESTRAINING ORDER', 'DISTURBING THE PEACE', 'RECKLESS DRIVING', 
        'PROWLER', 'DISRUPT SCHOOL', 'FAILURE TO DISPERSE', 'BLOCKING DOOR INDUCTION CENTER', 'INCITING A RIOT', 
        'FALSE POLICE REPORT', 'PEEPING TOM', 'BOMB SCARE'
    ],
    'KIDNAPPING & TRAFFICKING': [
        'KIDNAPPING', 'HUMAN TRAFFICKING',
        'CHILD STEALING', 'HUMAN TRAFFICKING - COMMERCIAL SEX ACTS', 'KIDNAPPING - GRAND ATTEMPT', 
        'FALSE IMPRISONMENT', 'HUMAN TRAFFICKING - INVOLUNTARY SERVITUDE'
    ],
    'OTHER / MISCELLANEOUS': [
        'OTHER OFFENSE', 'GAMBLING', 'RITUALISM', 'NON-CRIMINAL',
        'OTHER MISCELLANEOUS CRIME', 'LYNCHING', 'CRUELTY TO ANIMALS', 'CONSPIRACY', 'DRUNK ROLL', 
        'LYNCHING - ATTEMPTED', 'BIGAMY', 'TRAIN WRECKING', 'DRUNK ROLL - ATTEMPT'
    ]
}

# Reverse the dictionary to make it mapping-friendly for Pandas
# This creates a flat { 'OLD_VALUE': 'NEW_CATEGORY' } format
flat_mapping = {}
for new_category, old_values in unified_mapping.items():
    for old_val in old_values:
        # Using .strip() to clean up any hidden whitespace from raw datasets
        flat_mapping[old_val.strip()] = new_category

# Apply the mapping to both DataFrames

# Create a new column in Chicago dataset
chicago['Unified_Crime_Type'] = chicago['Primary Type'].str.strip().map(flat_mapping)
# Fill any missing/unmapped items with 'UNMAPPED' so they are easy to find
chicago['Unified_Crime_Type'] = chicago['Unified_Crime_Type'].fillna('UNMAPPED')

# Create a new column in LA dataset
LA['Unified_Crime_Type'] = LA['Crm Cd Desc'].str.strip().map(flat_mapping)
# Fill any missing/unmapped items with 'UNMAPPED'
LA['Unified_Crime_Type'] = LA['Unified_Crime_Type'].fillna('UNMAPPED')

# Check if any crimes slipped through the cracks
print("Unmapped in Chicago:", chicago[chicago['Unified_Crime_Type'] == 'UNMAPPED']['Primary Type'].unique())
print("Unmapped in LA:", LA[LA['Unified_Crime_Type'] == 'UNMAPPED']['Crm Cd Desc'].unique())

Unmapped in Chicago: []
Unmapped in LA: []


In [39]:
chicago.head(5)

,ID,Case Number,Date,Block,IUCR,Primary Type,Description,Location Description,Arrest,Domestic,...,Community Area,FBI Code,X Coordinate,Y Coordinate,Year,Updated On,Latitude,Longitude,Location,Unified_Crime_Type
0,12262588,JE104819,12/31/2020 11:59:00 PM,017XX N CICERO AVE,0810,THEFT,OVER $500,PARKING LOT / GARAGE (NON RESIDENTIAL),False,False,...,25.0,06,1144138.0,1910955.0,2020,2021 Jan 08 03:39:34 PM,41.911684,-87.745918,POINT (-87.745917518 41.9116839),THEFT / LARCENY
1,12258658,JE100080,12/31/2020 11:51:00 PM,013XX S HOMAN AVE,0430,BATTERY,AGGRAVATED - OTHER DANGEROUS WEAPON,STREET,False,False,...,29.0,04B,1153918.0,1893431.0,2020,2021 Jan 07 03:41:47 PM,41.863407,-87.710456,POINT (-87.710455868 41.863406877),ASSAULT & BATTERY
2,12274737,JE118886,12/31/2020 11:45:00 PM,001XX E WACKER DR,0810,THEFT,OVER $500,HOTEL / MOTEL,False,False,...,32.0,06,1177683.0,1902638.0,2020,2021 Jan 23 03:39:36 PM,41.888165,-87.622937,POINT (-87.622937212 41.888165132),THEFT / LARCENY
3,12258623,JD475015,12/31/2020 11:45:00 PM,056XX S WABASH AVE,0486,BATTERY,DOMESTIC BATTERY SIMPLE,APARTMENT,False,True,...,40.0,08B,1177678.0,1867609.0,2020,2021 Jan 07 03:41:47 PM,41.792043,-87.624018,POINT (-87.624017919 41.792043078),ASSAULT & BATTERY
4,12263896,JE105660,12/31/2020 11:45:00 PM,049XX S FORRESTVILLE AVE,0810,THEFT,OVER $500,RESIDENCE,False,False,...,38.0,06,1180760.0,1872402.0,2020,2021 Jan 10 03:39:57 PM,41.805125,-87.612569,POINT (-87.612569477 41.805125154),THEFT / LARCENY


In [40]:
LA.head(5)

,DR_NO,Date Rptd,DATE OCC,TIME OCC,AREA,AREA NAME,Rpt Dist No,Part 1-2,Crm Cd,Crm Cd Desc,...,Status Desc,Crm Cd 1,Crm Cd 2,Crm Cd 3,Crm Cd 4,LOCATION,Cross Street,LAT,LON,Unified_Crime_Type
0,211507896,2021 Apr 11 12:00:00 AM,11/07/2020 12:00:00 AM,845,15,N Hollywood,1502,2,354,THEFT OF IDENTITY,...,Invest Cont,354.0,NaN,NaN,NaN,7800 BEEMAN AV,NaN,34.2124,-118.4092,THEFT / LARCENY
1,201516622,2020 Oct 21 12:00:00 AM,10/18/2020 12:00:00 AM,1845,15,N Hollywood,1521,1,230,"ASSAULT WITH DEADLY WEAPON, AGGRAVATED ASSAULT",...,Invest Cont,230.0,NaN,NaN,NaN,ATOLL AV,N GAULT,34.1993,-118.4203,ASSAULT & BATTERY
2,240913563,2024 Dec 10 12:00:00 AM,10/30/2020 12:00:00 AM,1240,9,Van Nuys,933,2,354,THEFT OF IDENTITY,...,Invest Cont,354.0,NaN,NaN,NaN,14600 SYLVAN ST,NaN,34.1847,-118.4509,THEFT / LARCENY
3,210704711,2020 Dec 24 12:00:00 AM,12/24/2020 12:00:00 AM,1310,7,Wilshire,782,1,331,THEFT FROM MOTOR VEHICLE - GRAND ($950.01 AND ...,...,Invest Cont,331.0,NaN,NaN,NaN,6000 COMEY AV,NaN,34.0339,-118.3747,THEFT / LARCENY
4,201418201,2020 Oct 03 12:00:00 AM,09/29/2020 12:00:00 AM,1830,14,Pacific,1454,1,420,THEFT FROM MOTOR VEHICLE - PETTY ($950 & UNDER),...,Invest Cont,420.0,NaN,NaN,NaN,4700 LA VILLA MARINA,NaN,33.9813,-118.4350,THEFT / LARCENY


# Save Cleaned DataFrames Back to CSV
After cleaning the dataframes, we save them as cleaned CSV files to upload to GitHub since they will be much smaller after cleaning.

In [ ]:
# Save Chicago dataframe to a new file
chicago.to_csv('Chicago_Crime_Cleaned.csv', index=False)
print("Chicago cleaned CSV saved as 'Chicago_Crime_Cleaned.csv'.")

# Save LA dataframe to a new file
LA.to_csv('LA_Crime_Cleaned.csv', index=False)
print("LA cleaned CSV saved as 'LA_Crime_Cleaned.csv'.")

**Data Dictionary**

**Chicago Dataset**

**`ID`**

- `[ID]` a unique number associated with each crime as an identifier

**`Date`**

- `[Date]` the date the crime occurred, in month/day/year format, including the time

**`Latitude`**

- `[Latitude]` the latitude coordinate of the location where the crime occurred 
**`Longitude`**

- `[Longitude]` the longitude coordinate of the location where the crime occurred 

**`Crime Type`**

- `[Unified_Crime_Type]` the classification of the type of crime that occurred

**LA Dataset**

**`Division of Records Number`**

- `[DR_NO]` an identification number for each crime

**`Date Occurred`**

- `[DATE OCC]` the date in which the crime occurred in month/date/year format, including the time

**`Latitude`**

- `[LAT]` the latitude coordinate of the location where the crime occurred 

**`Longitude`**

- `[LON]` the longitude coordinate of the location where the crime occurred 
